### PR TITLE
fix: split publish errors to separate lines

### PR DESCRIPTION
**Pull request by myishay**

### CODE REVIEW

File: cmd/publish/main.go [hunk 1]
```diff
@@ -80,17 +89,33 @@ type MessagesContext struct {
...
 	policiesConfiguration, err := files.ExtractYamlFileToUnknownStruct(path)
 	if err != nil {
-		return err
+		return PublishFailedResponse{}, err
 	}
 
-	err = ctx.PublishCliClient.PublishPolicies(policiesConfiguration, localConfigContent.CliId)
-	return err
+	res, publishErr := ctx.PublishCliClient.PublishPolicies(policiesConfiguration, localConfigContent.CliId)
+	if publishErr != nil {
```
Comment (dimabru) [inline cmd/publish/main.go]: return `publishErr` if exists

**Review by dimabru — DISMISSED** [review]

File: cmd/publish/main.go [hunk 2]
```diff
@@ -22,7 +25,7 @@ type LocalConfig interface {
 }
 
 type CliClient interface {
-	PublishPolicies(policiesConfiguration files.UnknownStruct, cliId string) error
+	PublishPolicies(policiesConfiguration files.UnknownStruct, cliId string) (httpClient.Response, error)
```
Comment (noaabarki) [inline cmd/publish/main.go]: Let's return a pointer `*httpClient.Response`

Comment (noaabarki) [reply cmd/publish/main.go]: I'm pretty sure the CliClient shouldn't be exported or used outside of publish package. If I'm not mistaken, let's use `cliClient` instead

File: cmd/publish/main.go [hunk 3]
```diff
@@ -56,9 +59,15 @@ func New(ctx *PublishCommandContext) *cobra.Command {
 				}
 			}()
 
-			err := publish(ctx, args[0])
-			if err != nil {
-				ctx.Printer.PrintMessage("Publish failed:\n"+err.Error()+"\n", "error")
+			publishFailedResponse, err := publish(ctx, args[0])
+			if len(publishFailedResponse.Payload) > 0 {
+				for _, message := range publishFailedResponse.Payload {
+					ctx.Printer.PrintMessage("Publish failed:\n", "error")
```
Comment (noaabarki) [inline cmd/publish/main.go]: Why twice `"Publish failed:\n", "error"`?

File: cmd/publish/main.go [hunk 4]
```diff
@@ -80,17 +89,34 @@ type MessagesContext struct {
 	CliClient   *cliClient.CliClient
 }
 
-func publish(ctx *PublishCommandContext, path string) error {
+type PublishFailedResponse struct {
```
Comment (noaabarki) [inline cmd/publish/main.go]: The same comment on the `CliClient` -> `cliClient`

File: cmd/publish/main.go [hunk 5]
```diff
@@ -80,17 +89,34 @@ type MessagesContext struct {
...
+	Code    string   `json:"code"`
+	Message string   `json:"message"`
+	Payload []string `json:"payload"`
+}
+
+func publish(ctx *PublishCommandContext, path string) (PublishFailedResponse, error) {
```
Comment (noaabarki) [inline cmd/publish/main.go]: Please return pointer instead of new value

File: pkg/cliClient/publish.go [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
 import (
-	"github.com/datreeio/datree/bl/files"
 	"net/http"
+
+	"github.com/datreeio/datree/bl/files"
+	"github.com/datreeio/datree/pkg/httpClient"
 )
 
-func (c *CliClient) PublishPolicies(policiesConfiguration files.UnknownStruct, cliId string) error {
+func (c *CliClient) PublishPolicies(policiesConfiguration files.UnknownStruct, cliId string) (httpClient.Response, error) {
```
Comment (noaabarki) [inline pkg/cliClient/publish.go]: 👆🏼

File: pkg/cliClient/cliClient_test.go [hunk 7]
```diff
@@ -4,12 +4,13 @@ import (
...
 	"encoding/json"
 	"errors"
-	"github.com/datreeio/datree/bl/files"
 	"io/ioutil"
 	"net/http"
 	"path/filepath"
 	"testing"
 
+	"github.com/datreeio/datree/bl/files"
+
```
Comment (noaabarki) [inline pkg/cliClient/cliClient_test.go:13]: Do the tests cover what you added?

**Review by noaabarki — CHANGES_REQUESTED** [review]

**Pull request merged**